ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncActions.java

```diff
@@ -131,11 +130,17 @@ public CompletableFuture<FastSyncState> downloadPivotBlockHeader(
   private CompletableFuture<FastSyncState> internalDownloadPivotBlockHeader(
       final FastSyncState currentState) {
     if (currentState.hasPivotBlockHeader()) {
-      LOG.info("Initial sync state {} already contains the block header", currentState);
+      LOG.debug("Initial sync state {} already contains the block header", currentState);
       return completedFuture(currentState);
     }
 
-    return waitForPeers(1)
+    return ethContext
+        .getEthPeers()
+        .waitForPeer(
+            (peer) ->
+                currentState.getPivotBlockNumber().isEmpty()
+                    || peer.chainState().getEstimatedHeight()
```

Comment:
We can't rely on the estimated height in pos networks as peers don't send the new block hash message in those networks. So either we guard this on POS or just remove the estimated height check here.

Reply:
This check has been removed, so I'll focus discussion on the other related comment